src/harvest-rpde.js

```diff
@@ -0,0 +1,205 @@
+              ...progressFromContext(context),
+              status: context.items === 0 ? 'Harvesting Complete (No items to validate)' : 'Harvesting Complete, Validating...',
+            });
+            context._progressbar.setTotal(context.totalItemsQueuedForValidation);
+          }
+          isInitialHarvestComplete = true;
+        }
+        if (WAIT_FOR_HARVEST || VALIDATE_ONLY) {
+          await onFeedEnd();
+        } else if (VERBOSE) log(`Sleep mode poll for RPDE feed "${url}"`);
+        context.sleepMode = true;
+        if (context.timeToHarvestCompletion === undefined) context.timeToHarvestCompletion = millisToMinutesAndSeconds((new Date()).getTime() - startTime.getTime());
+        // Slow down sleep polling while waiting for harvesting of other feeds to complete
+        await sleep(WAIT_FOR_HARVEST && incompleteFeeds.length !== 0 ? 5000 : 500);
+      } else {
+        context.responseTimes.push(responseTime);
+        // Maintain a buffer of the last 5 items
+        if (context.responseTimes.length > 5) context.responseTimes.shift();
+        context.pages += 1;
+        context.items += json.items.length;
+        delete context.sleepMode;
+        if (REQUEST_LOGGING_ENABLED) {
+          const kind = json.items && json.items[0] && json.items[0].kind;
+          log(
+            `RPDE kind: ${kind}, page: ${context.pages}, length: ${json.items.length
+            }, next: '${json.next}'`,
+          );
+        }
+        // eslint-disable-next-line no-loop-func
+        await processPage(json, feedContextIdentifier, () => isInitialHarvestComplete);
+        if (!isInitialHarvestComplete && context._progressbar) {
+          context._progressbar.update(context.validatedItems, {
+            pages: context.pages,
+            responseTime: Math.round(responseTime),
+            ...progressFromContext(context),
+          });
+          context._progressbar.setTotal(context.totalItemsQueuedForValidation);
+        }
+        url = json.next;
+      }
+      numberOfRetries = 0;
+    } catch (error) {
+      // Do not wait for the Orders feed if failing (as it might be an auth error)
+      if ((WAIT_FOR_HARVEST || VALIDATE_ONLY) && isOrdersFeed) {
+        onFeedEnd();
+      }
+      if (error.name === 'FatalError') {
+        // If a fatal error, quit the application immediately
+        if (multibar) multibar.stop();
+        logError(`\nFATAL ERROR for RPDE feed ${feedContextIdentifier} page "${url}": ${error.message}\n`);
+        process.exit(1);
+      } else if (!error.isAxiosError) {
+        // If a non-axios error, quit the application immediately
+        if (multibar) multibar.stop();
+        logErrorDuringHarvest(`FATAL ERROR for RPDE feed ${feedContextIdentifier} page "${url}": ${error.message}\n${error.stack}`);
+        process.exit(1);
+      } else if (error.response?.status === 404) {
+        // If 404, simply stop polling feed
+        if (WAIT_FOR_HARVEST || VALIDATE_ONLY) { await onFeedEnd(); }
```

Comment:
`onFeedEnd()` could be called twice here if its an Orders feed and the error is a 404 error